### PR TITLE
feat: prevent accessing PHP variables

### LINKED ISSUE #2686: Detect `$ {` PHP injection
### Description

In #2668, it was shown that PHP variables can be accessed by `$ {` (note the space between the chars).

As @theMiddleBlue found, PL1 can be bypassed for example:

```
print_r($ {'_SER'.'VER'});
```

We have a rule 932130, which fires on `${` to detect unix shell snippets. I checked but in the shell, adding a space between `$` and `{` doesn't work:

```
# testy=123
# echo ${testy}
123
# echo $ {testy}
$ {testy}
```

But in PHP it works. So the right place to detect this, is not the rule 932130, but a new rule, with a different attack class: PHP code injection. 

### Your Environment

* CRS version (e.g., v3.2.0): nightly
* Paranoia level setting: 2
* ModSecurity version (e.g., 2.9.3): 2.9.3
* Web Server and version (e.g., apache 2.4.41): apache
* Operating System and version: n/a

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

### PR DESCRIPTION
**Pull request by azurit**
PHP allows accessing variables using method `${VAR}` which is currently not catched by any rule. This PR adds support for it.

Fixes #2686.

### CODE REVIEW

**Review by fzipi — COMMENTED** [review]
Thanks for taking this to the finish line!

At a first look, it is sound. Did checked to see if the regex could be "problematic", but looks safe.

All the tests I did (besides the one you wrote) worked.

No additional false positives from QT, so we should be good.

What to do you think about other potential problems? I did a quick review on XML but I think we are also safe (looked at XPATH queries, etc.)

If I'm not missing something critical, I think we can merge.

Just in case, let me involve also @theseion for a quick review.

**Comment by azurit** [comment]
I was thinking only about the unlimited size of the variable name (or a pattern between `{` and `}`). Would be cool if it's possible to limit it to at least to some size but, unfortunately, PHP has no lenght limit for variable names.

**Review by theseion — DISMISSED** [review]

File: rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf [hunk 1]
```diff
@@ -161,6 +161,37 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
...
+# [ PHP Variables ]
+#
+# Prevent accessing PHP variables using these methods:
+# ${'VARIABLE_NAME'}
+# $ {"VARIABLE_NAME"}
+# $ {'_VAR'.'IABLE_NAME'}
+# $     {   $var}
+# $     {   CONSTANT   }
+#
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_FILENAME|ARGS_NAMES|ARGS|XML:/* "@rx \$ *\{.+\}" \
```
[the remark this answers is not in the document]
Comment (azurit) [reply rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf]: Yes, true, thanks, but i added only the first `\s`. The rest of your change does not have much sense.

[the remark this answers is not in the document]
Comment (azurit) [reply rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf]: Ok, now i see it. But also `+` must be changed to `*`.

**Comment by azurit** [comment]
Added one more test (single character variable name).

**Review by theseion — COMMENTED** [review]
LGTM. Still vulnerable to ReDoS. What's your take @fzipi?

**Comment by fzipi** [comment]
I don't think PHP will work with nested braces, right? 

If that's the case, I think it is better to use `\$\s*\{\s*\S[^{}]*\}` instead?

**Comment by azurit** [comment]
> I don't think PHP will work with nested braces, right?
> 
> If that's the case, I think it is better to use `\$\s*\{\s*\S[^{}]*\}` instead?

@fzipi If you mean this then it is working:
`${${var}}`

But it may be sufficiet to match only the nested part.

**Comment by fzipi** [comment]
Exactly, matching the nested part will solve it for us.

**Review by fzipi — APPROVED** [review]
Looks like this solves our problems! Thanks @azurit !

File: tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933135.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            Proxy-Connection: keep-alive
+            User-Agent: "OWASP CRS test agent"
+          method: GET
+          port: 80
+          uri: "/get?foo=%24%20%20%20%20%20%7B%20%20%20CONSTANT%20%20%20%7D"
+          version: HTTP/1.1
+        output:
+          log:
+            expect_ids: [933135]
+  - test_id: 7
```
[the remark this answers is not in the document]
Comment (fzipi) [reply tests/regression/tests/REQUEST-933-APPLICATION-ATTACK-PHP/933135.yaml:126]: Test 8 added.

**Pull request merged**